client/src/reducers/leadsReducer.js

```diff
@@ -11,5 +11,19 @@ export default function(
         leads: action.payload
       };
   }
+  switch (action.type) {
+    case 'GET_ALL_LEADS_COLUMNS':
+    return {
+      ...state,
+      leadsColums: action.payload
```

Comment:
Did you intentionally name the property leadsColums (without an 'n' at end of column)?   All other properties and variable names you use (outside of reducer and mapStateToProps) that contain the word column spell it correctly, with an N at the end.